tests/src/test/spark350/scala/com/nvidia/spark/rapids/ToPrettyStringSuite.scala

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+  }
+
+  test("test show() Struct") {
+    val dataType = StructType(Seq(StructField("a", DataTypes.IntegerType),
+      StructField("b", DataTypes.IntegerType),
+      StructField("c", DataTypes.IntegerType)))
+    val v = Array(1, 2, 3)
+    val cpuResult = executeOnCpuAndReturn(dataType, InternalRow(v: _*))
+    val child = GpuBoundReference(0, dataType, true)(NamedExpression.newExprId, "struct_arg")
+    val gpuToPrettyStr = GpuToPrettyString(child, Some("UTC"))
+    val l =
+      List[HostColumnVector.DataType](
+        new HostColumnVector.BasicType(false, DType.INT32),
+        new HostColumnVector.BasicType(false, DType.INT32),
+        new HostColumnVector.BasicType(false, DType.INT32)
+      ).asJava
+    val structType = new HostColumnVector.StructType(false, l)
+    withResource(
+      GpuColumnVector.from(
+        ColumnVector.fromStrings(cpuResult), DataTypes.StringType)) { expected =>
+      withResource(GpuColumnVector.from(
+        ColumnVector.fromStructs(structType,
+          List(new HostColumnVector.StructData(v.map(Int.box): _*)).asJava
+        ), dataType)) { input =>
+        val batch = new ColumnarBatch(List(input).toArray, 1)
+        checkEvaluation(gpuToPrettyStr, expected, batch)
+      }
+    }
+  }
+
+  private def testDataType(dataType: DataType): Unit = {
+    val schema = (new StructType)
+      .add(StructField("a", dataType, true))
+    val numRows = 100
+    val inputRows = GpuBatchUtilsSuite.createRows(schema, numRows)
+    val cpuOutput: Array[String] = inputRows.map {
+      input => 
+        ToPrettyString(BoundReference(0, dataType, true), Some("UTC"))
+        .eval(input).asInstanceOf[UTF8String].toString()
+    }
+    val child = GpuBoundReference(0, dataType, true)(NamedExpression.newExprId, "arg")
+    val gpuToPrettyStr = GpuToPrettyString(child, Some("UTC"))
+
+    withResource(new GpuColumnarBatchBuilder(schema, numRows)) { batchBuilder =>
+      val r2cConverter = new GpuRowToColumnConverter(schema)
+      inputRows.foreach(r2cConverter.convert(_, batchBuilder))
+      withResource(batchBuilder.build(numRows)) { columnarBatch =>
+        withResource(GpuColumnVector.from(ColumnVector.fromStrings(cpuOutput: _*),
+          DataTypes.StringType)) { expected =>
+          checkEvaluation(gpuToPrettyStr, expected, columnarBatch)
+        }
+      }
+    }
+  }
+
+  test("test strings") {
+    testDataType(DataTypes.StringType)
+  }
+
+  test("test floats") {
+    // This test is expected to fail until https://github.com/NVIDIA/spark-rapids/issues/4204
+    // is resolved
+    assertThrows[TestFailedException](testDataType(DataTypes.FloatType))
+  }
+
+  test("test doubles") {
+    // This test is expected to fail until https://github.com/NVIDIA/spark-rapids/issues/4204
+    // is resolved
+    assertThrows[TestFailedException](testDataType(DataTypes.DoubleType))
+  }
+
+  test("test ints") {
+    testDataType(DataTypes.IntegerType)
+  }
+
+  test("test longs") {
```

Comment:
Could you add tests for the remaining primitive types: char, byte, short, boolean

Reply:
I have added tests can you PTAL